gapir/client/client.go

```diff
@@ -102,33 +140,152 @@ func (client *Client) Connect(ctx context.Context, device bind.Device, abi *devi
 
 	log.I(ctx, "Waiting for connection to GAPIR...")
 
-	connection, err := newConnection(fmt.Sprintf("localhost:%d", newDeviceConnectionInfo.port), newDeviceConnectionInfo.authToken, connectTimeout)
+	// Create gRPC connection
+	conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", newDeviceConnectionInfo.port),
```

Comment:
I don't know if it is a good idea to call grpc directly here. It seems to be violation of separation of concerns. For example, if we would want to test this function and want to use a mock instead of grpc, we cannot do it.

Reply:
In the case of gapir, we can create a mock GAPIR gRPC server, and pass its listening `port` to this piece of code. The mock creation would actually happen in `device_connection.go`.